### ISSUE TITLE
Disable a cop for a specific directory

### ISSUE THREAD

**Issue by lanej** [bbatsov/rubocop#3694]
## Expected behavior

I would like to be able to disable a cop for a specific directory.  This is an issue arose from working with minitest and bumping into `Metrics/BlockLength` violations.  I want that cop disabled for `test/` but enabled for everything else.

## Actual behavior

Currently a cop is either on or off for all files and folders.  I can exclude the `test/` directory entirely but that disables all cops and I just want to disable one cop.

## Steps to reproduce the problem

* Create a `test` directory that contains a file with a `Metrics/BlockLength` violation and another violation.
* Create a `lib` directory that contains a file with a block that violates the maximum number of lines.
* I want to exclusively see the non-`Metrics/BlockLength` violation in `test/` and the `Metrics/BlockLength` violation in `lib`.

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
[0] » rubocop -V
0.45.0 (using Parser 2.3.1.4, running on ruby 2.3.1 x86_64-linux)
```

**Comment by rrosenblum** [comment]
This is already something that is supported by RuboCop.

Add this to your project's `.rubocop.yml` file.
```yml
Metrics/BlockLength:
  Enabled: true
  Exclude: 
    - test/**/*
```

**Comment by lanej** [comment]
@rrosenblum excellent! thank you.

**Comment by jdickey** [comment]
Just to chime in, shouldn't adding a new `.rubocop.yml` file in the directory of interest (here, `test/`), with

```
Metrics/BlockLength:
  Enabled: false
```

have the same effect? We're currently debating which usage is preferable: a Single Point of Truth should mean "no surprises", but at the cost of a (potentially) unwieldy process of flipping through `Exclude:` directives to see which if any apply to the file/directory that's confounding you at the moment.

Thoughts on that?

**Comment by lanej** [comment]
@jdickey we have a 'Single Point of Truth' and it works well.  A nice side effect is that it standardizes some directory structures.

I would prefer looking at one file over searching the file tree and mentally determining the inheritance.

**Comment by rrosenblum** [comment]
@jdickey, I agree with @lanej. I think there is good value in maintaining most/all of the configuration in a single top level `.rubocop.yml`. Both ways are supported because everyone will have their own opinions on the topic. 

One scenario that I can think of which would require multiple `.rubocop.yml` files is if you wanted to have different configuration values for a cop in a directory. For example, rather than disabling `Metrics/BlockLength` for the `test` directory, the `test` directory could have a `.rubocop.yml` file that changes the `Max` value of `Metrics/BlockLength`. As far as I know, changing the value of `EnforcedStyle` can only be done with separate configuration files.

**Comment by jdickey** [comment]
@rrosenblum Agreed, which is why I gave a 👍  to @lanej. My own personal feeling aligns with what I understand of yours: 

* enable/disable cops per directory in a master `.rubocop.yml`;
* set project-wide defaults there;
* use per-directory `.rubocop.yml` files for experimentation; when you find what works, then
* keep "approved" per-directory settings in the top-level `.rubocop.yml`

We had one project a while back that would rewrite subdirectory-level `.rubocop.yml` files *based on tags in the most recent commit* of each of the files in that directory. We're much better now, thanks.

**Comment by scottmatthewman** [comment]
We generally use a mixture of styles: Our project standards all sit within `PROJECT_ROOT/.rubocop.yml`, which by default just inherits from our central house style gem.

However, we have additional RSpec-specific settings in `PROJECT_ROOT/spec/.rubocop.yml`, because we have much more relaxed rules about method length, line length, etc. and it helps to have all those differences in just the one place. The `spec/` rubocop file inherits from the project file, and from the house style gem's RSpec defaults.

At the moment, we also have a separate settings file for Rails apps' `db` folder, but I'm in the process of phasing that out.
